runtime/parachains/src/scheduler.rs

```diff
@@ -36,135 +36,46 @@
-	queue: Vec<QueuedParathread>,
-	// this value is between 0 and config.parathread_cores
-	next_core_offset: u32,
-}
-
-impl ParathreadClaimQueue {
-	/// Queue a parathread entry to be processed.
-	///
-	/// Provide the entry and the number of parathread cores, which must be greater than 0.
-	fn enqueue_entry(&mut self, entry: ParathreadEntry, n_parathread_cores: u32) {
-		let core_offset = self.next_core_offset;
-		self.next_core_offset = (self.next_core_offset + 1) % n_parathread_cores;
-
-		self.queue.push(QueuedParathread { claim: entry, core_offset })
-	}
-
-	/// Take next queued entry with given core offset, if any.
-	fn take_next_on_core(&mut self, core_offset: u32) -> Option<ParathreadEntry> {
-		let pos = self.queue.iter().position(|queued| queued.core_offset == core_offset);
-		pos.map(|i| self.queue.remove(i).claim)
-	}
-
-	/// Get the next queued entry with given core offset, if any.
-	fn get_next_on_core(&self, core_offset: u32) -> Option<&ParathreadEntry> {
-		let pos = self.queue.iter().position(|queued| queued.core_offset == core_offset);
-		pos.map(|i| &self.queue[i].claim)
-	}
-}
-
-impl Default for ParathreadClaimQueue {
-	fn default() -> Self {
-		Self { queue: vec![], next_core_offset: 0 }
-	}
-}
-
-/// Reasons a core might be freed
-#[derive(Clone, Copy)]
-pub enum FreedReason {
-	/// The core's work concluded and the parablock assigned to it is considered available.
-	Concluded,
-	/// The core's work timed out.
-	TimedOut,
-}
-
-/// The assignment type.
-#[derive(Clone, Encode, Decode, TypeInfo)]
-#[cfg_attr(feature = "std", derive(PartialEq, Debug))]
-pub enum AssignmentKind {
-	/// A parachain.
-	Parachain,
-	/// A parathread.
-	Parathread(CollatorId, u32),
-}
-
-/// How a free core is scheduled to be assigned.
-#[derive(Clone, Encode, Decode, TypeInfo)]
-#[cfg_attr(feature = "std", derive(PartialEq, Debug))]
-pub struct CoreAssignment {
-	/// The core that is assigned.
-	pub core: CoreIndex,
-	/// The unique ID of the para that is assigned to the core.
-	pub para_id: ParaId,
-	/// The kind of the assignment.
-	pub kind: AssignmentKind,
-	/// The index of the validator group assigned to the core.
-	pub group_idx: GroupIndex,
-}
-
-impl CoreAssignment {
-	/// Get the ID of a collator who is required to collate this block.
-	pub fn required_collator(&self) -> Option<&CollatorId> {
-		match self.kind {
-			AssignmentKind::Parachain => None,
-			AssignmentKind::Parathread(ref id, _) => Some(id),
-		}
-	}
-
-	/// Get the `CoreOccupied` from this.
-	pub fn to_core_occupied(&self) -> CoreOccupied {
-		match self.kind {
-			AssignmentKind::Parachain => CoreOccupied::Parachain,
-			AssignmentKind::Parathread(ref collator, retries) =>
-				CoreOccupied::Parathread(ParathreadEntry {
-					claim: ParathreadClaim(self.para_id, collator.clone()),
-					retries,
-				}),
-		}
-	}
-}
+const LOG_TARGET: &str = "runtime::parachains::scheduler";
+pub mod migration;
 
 #[frame_support::pallet]
 pub mod pallet {
 	use super::*;
```

Comment:
This seems to be a duplicate of 

https://github.com/paritytech/polkadot/blob/5bda9b65fd0d17c99d81ced0cbdba6294dc8a2e0/runtime/parachains/src/scheduler/migration.rs#L88

Better keep the version in a single place.